File: leetcode2/0532. K-diff Pairs in an Array.py
class Solution:
    def findPairs(self, nums: 'List[int]', k: 'int') -> 'int':
        dictnums={}
        for num in nums:
            if num not in dictnums:
                dictnums[num]=1
            else:
                dictnums[num]+=1
        rep=0
        if k<0:
            return 0
        if k==0:
            for key in dictnums:
                if dictnums[key]>1:
                    rep+=1
        else:
            for key in dictnums:
                if key+k in dictnums:
                    rep+=1
        return rep

File: C#/0532. K-diff Pairs in an Array.cs
public class Solution {
    public int FindPairs(int[] nums, int k) {
        Array.Sort(nums);
        int left=0;
        int right=1;
        int rep=0;
        while(right<nums.Length){
            if(nums[right]-nums[left]>k){
                left++;
            }
            else if(nums[right]-nums[left]<k){
                right++;
            }
            else{
                rep++;
                int targetLeft=nums[left];
                while(left<nums.Length && nums[left]==targetLeft){
                    left++;
                }
            }
            if(left>=right){
                right=left+1;
            }
        }
        return rep;
    }
}
